frontend/src/pages/projects/screens/spawner/storage/utils.ts

```diff
@@ -22,35 +22,33 @@ export const useCreateStorageObject = (
 ): [
   data: StorageData,
   setData: UpdateObjectAtPropAndValue<StorageData>,
-  resetDefaults: () => void,
 ] => {
   const size = useDefaultPvcSize();
-  const createDataState = useGenericObjectState<StorageData>({
-    name: '',
-    description: '',
-    size,
-  });
-  const [, setCreateData] = createDataState;
 
   const existingName =
     formData?.name || (existingData ? getDisplayNameFromK8sResource(existingData) : '');
+  const existingK8sName = formData?.k8sName || (existingData ? existingData.metadata.name : '');
   const existingDescription =
     formData?.description || (existingData ? getDescriptionFromK8sResource(existingData) : '');
   const existingSize =
     formData?.size || (existingData ? existingData.spec.resources.requests.storage : size);
   const existingStorageClassName =
     formData?.storageClassName || existingData?.spec.storageClassName;
 
-  React.useEffect(() => {
-    if (existingName) {
-      setCreateData('name', existingName);
-      setCreateData('description', existingDescription);
-      setCreateData('size', existingSize);
-      setCreateData('storageClassName', existingStorageClassName);
-    }
-  }, [existingName, existingDescription, setCreateData, existingSize, existingStorageClassName]);
+  const existingState = React.useMemo(
+    () => ({
+      name: existingName,
+      k8sName: existingK8sName,
+      description: existingDescription,
+      size: existingSize,
+      storageClassName: existingStorageClassName,
+    }),
+    [existingName, existingK8sName, existingDescription, existingSize, existingStorageClassName],
```

Comment:
Using a memo here has no value because it is simply used as the default in `useGenericObjectState` and then saved a ref internally.